go/tricorder/metric_test.go

```diff
@@ -1737,6 +1775,16 @@ func TestListNoChangeSubType(t *testing.T) {
 	t.Error("panic expected")
 }
 
+func TestNoAssignedUnitNoAdd(t *testing.T) {
+	bucketer := NewGeometricBucketer(1, 1000)
+	dist := newDistribution(bucketer, false)
+	defer func() {
+		recover()
```

Comment:
You can define an Err* variable which has the define error string, pass that to panic() and check that recover() returned that.